### PR TITLE
servermaster,executor(engine): Enable local file resource GC

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: ref #5513 

### What is changed and how it works?
- Setting up required components in both Servermaster and Executor.
- Made Executor's local data path configurable.
- e2e test will be added in another PR.

#### Tests 

 - Unit test
 - Integration test (only no regression is guaranteed)

##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: engine/servermaster/server.go [hunk 1]
```diff
@@ -748,6 +754,24 @@ func (s *Server) runLeaderService(ctx context.Context) (err error) {
...
 
+	s.gcRunner = externRescManager.NewGCRunner(s.frameMetaClient, map[resModel.ResourceType]externRescManager.GCHandlerFunc{
+		"local": resourcetypes.NewLocalFileResourceType(clients).GCHandler(),
+	})
+	s.gcCoordinator = externRescManager.NewGCCoordinator(s.executorManager, s.jobManager, s.frameMetaClient, s.gcRunner)
+
+	errg, errgCtx := errgroup.WithContext(ctx)
+	defer func() {
+		err := errg.Wait()
+		log.L().Error("errgroup exited with error", zap.Error(err))
```
Comment (amyangfei) [inline engine/servermaster/server.go:765]: Is it better to check `err` is not nil and then log error.

**Comment by amyangfei** [comment]
And please resolve conflicts, rest lgtm

File: engine/executor/config.go [hunk 2]
```diff
@@ -216,3 +216,10 @@ func (c *Config) configFromFile(path string) error {
 	}
 	return nil
 }
+
+func getDefaultLocalStorageDir(executorName string) string {
+	// Use hex encoding in case there are special characters in the
+	// executor name.
+	encodedExecutorName := hex.EncodeToString([]byte(executorName))
+	return filepath.Join(defaultLocalStorageDirPrefix, encodedExecutorName)
```
Comment (lance6716) [inline engine/executor/config.go:224]: Do we need to check existence of the path, or use ioutil.TempDir to make sure always use a new directory?

**Review by lance6716 — APPROVED** [review]

**Review by amyangfei — APPROVED** [review]

**Pull request merged**